galaxy_importer/schema.py

```diff
@@ -305,22 +305,72 @@ def _check_license_or_license_file(self):
             f"'license_file' ({self.license_file}) were invalid.")
 
 
+@attr.s(frozen=True)
+class CollectionArtifactFile(object):
+    name = attr.ib()
+    ftype = attr.ib()
+    src_name = attr.ib(default=None)
+    chksum_type = attr.ib(default="sha256")
+    chksum_sha256 = attr.ib(default=None)
+    format = attr.ib(default=1)
+
+
 @attr.s(frozen=True)
 class CollectionArtifactManifest(object):
     """Represents collection manifest metadata."""
 
     collection_info = attr.ib(type=CollectionInfo)
     format = attr.ib(default=1)
-    file_manifest_file = attr.ib(factory=dict)
+
+    file_manifest_file = attr.ib(default=None,
```

Comment:
> can we separate the analysis of `FILES.json` and `MANIFEST.json`?

ditto

Reply:
Not sure I follow. By analysis do you mean the attr validators?

Reply:
the `CollectionArtifactManifest` class is all about `MANIFEST.json`, this edit now makes it reference the `FILES.json`

Reply:
file_manifest_file was always the section of Manifest.json that describes FILES.json.
Prev it was just a dict datastructure, but now it is a CollectionArtifactFile instance.